### PR TITLE
 rgw: make radosgw-admin user create and modify distinct

### PR DESCRIPTION
**Pull request by matthewoliver**
Currently if you run 'radosgw-admin user create ..' when the user
already exists and you happen to specify, at least,  '--uid' and
'--display-name' that match the existing user, radowgw-admin will
actaully go modify the existing user.

This behaviour is a little confusing, hence the bug this patch is
fixing. This patch instead simplifies the tool to make
'create' create and 'modify' modify.
Meaning when you go 'create' a user that already exists, you'll get an
error, as expected. If you want to modify a user, you actually have to
use 'modify'.

For exapmle, now:

  $ radosgw-admin user create --uid="test-user"  --display-name="test user"
  could not create user: unable to create user, user: test-user exists

Signed-off-by: Matthew Oliver <moliver@suse.com>
Fixes: https://tracker.ceph.com/issues/38619

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by matthewoliver** [comment]
Just a rebase to keep it current

File: src/rgw/rgw_user.cc [hunk 1]
```diff
@@ -1706,7 +1706,10 @@ int RGWUser::execute_add(RGWUserAdminOpState& op_state, std::string *err_msg)
     if (!op_state.exclusive &&
         (user_email.empty() ||
 	 boost::iequals(user_email, old_info.user_email)) &&
-        old_info.display_name == display_name) {
+        old_info.display_name == display_name &&
+        op_state.will_modify_user_on_create()) {
+      cout << "unable to create user, user: " << op_state.user_id.to_str()
```
Comment (dang) [inline src/rgw/rgw_user.cc]: This probably needs to be cerror not cout, since the result is parsable json, and we don't want to confuse the parser.

**Comment by matthewoliver** [comment]
To be honest, I always thought it strange that an add could become a modify. I'd rather error with a user exists, and then force someone to use modify directly. But being new to the code base and seeing clearly in the code that this is obviously intentional  figured, either it was expected behaviour or something legacy that we need to keep as there might be tools out there that expect this contract.

But perfectly happy to clean up the code to make 'add' add and 'modify' modify. Makes much more sense to me.

**Comment by dang** [comment]
Awesome, let's do that.

**Comment by matthewoliver** [comment]
OK, here is a new version that just makes 'create' and 'modify' commands distinct. It's a much easier patch.

**Comment by cbodley** [comment]
jenkins test signed

**Comment by dang** [comment]
http://pulpito.ceph.com/dang-2020-01-28_08:08:31-rgw-wip-dang-testing-distro-basic-smithi/

**Comment by dang** [comment]
jenkins test signed

**Comment by dang** [comment]
@matthewoliver This is ready to merge, but Jenkins doesn't like the signed off line.  I think it wants the email to be in <>?

**Comment by matthewoliver** [comment]
Thanks @dang good spot. Fixed (hopefully) the signed-off-by

**Pull request merged**